### PR TITLE
Remove from numpy cimport *

### PR DESCRIPTION
**Pull request by jbrockmendel**
Replace these imports with explicit imports.

There are a couple of other similar things coming up, splitting them apart for easier review.

There are a couple of places with `cdef extern from "../src/headers/math.h":` but I don't think the path here makes sense.  The fact that it works anyway bothers me.  This PR changes one such occurrence as a demonstration, leaves the others for the next PR.

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/_libs/algos.pyx [hunk 1]
```diff
@@ -15,31 +15,19 @@ cimport util
...
+                    double_t)
+
 
 cdef double NaN = <double> np.NaN
 cdef double nan = NaN
 
-cdef extern from "../src/headers/math.h":
-    double sqrt(double x) nogil
-    double fabs(double) nogil
+from cpython.math cimport sqrt, fabs
```
Comment (chris-b1) [inline pandas/_libs/algos.pyx]: I think you want `libc.math` here?

File: pandas/_libs/groupby.pyx [hunk 2]
```diff
@@ -1,16 +1,17 @@
 # cython: profile=False
 
-from numpy cimport *
 cimport numpy as np
```
Comment (chris-b1) [inline pandas/_libs/groupby.pyx]: we're inconsistent about this, but at least I prefer the `cimport numpy as np`, so it is clearer which parts need c-level access

File: pandas/_libs/reshape.pyx [hunk 3]
```diff
@@ -1,34 +1,19 @@
...
-from numpy cimport NPY_INT16 as NPY_int16
-from numpy cimport NPY_INT32 as NPY_int32
-from numpy cimport NPY_INT64 as NPY_int64
-from numpy cimport NPY_FLOAT16 as NPY_float16
-from numpy cimport NPY_FLOAT32 as NPY_float32
-from numpy cimport NPY_FLOAT64 as NPY_float64
-
-from numpy cimport (int8_t, int16_t, int32_t, int64_t, uint8_t, uint16_t,
+from numpy cimport (ndarray,
+					int8_t, int16_t, int32_t, int64_t, uint8_t, uint16_t,
```
Comment (chris-b1) [inline pandas/_libs/reshape.pyx]: formatting

**Review by chris-b1 — CHANGES_REQUESTED** [review]

File: pandas/_libs/algos.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/algos.pyx]: Good catch.  Just pushed a fix.

File: pandas/_libs/reshape.pyx (hunk 3 above)
Comment (jbrockmendel) [reply pandas/_libs/reshape.pyx]: Fixed.

File: pandas/_libs/groupby.pyx (hunk 2 above)
Comment (jreback) [reply pandas/_libs/groupby.pyx]: agree here

Comment (jbrockmendel) [reply pandas/_libs/groupby.pyx]: Coming right up.

**Comment by jreback** [comment]
looks ok to me. @chris-b1

**Review by chris-b1 — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**